### DISCUSSION TITLE
Docker and MySQL deprecation schedules

### DISCUSSION THREAD

**Discussion by bhenc** [quay/quay#3172] [Q&A] [answered]
Greetings,

The project quay documentation [mentions](https://docs.projectquay.io/deploy_quay_ha.html) that docker will be deprecated in a future release:
> Podman is strongly recommended for highly available, production quality deployments of Project Quay. Docker has not been tested with Project Quay 3.10, and will be deprecated in a future release.

Could you please elaborate on when docker support will be removed and what the thinking is behind the decision?

Similar question regarding MySQL support deprecation, it would help to know by which release it will stop being supported.

Thank you in advance.

**Comment by dmesser** [comment, marked as answer]
@bhenc This is an artifact of the official Red Hat Quay product documentation. It does not apply to the Project Quay FOSS, since it comes without "support" anyway. Technically, we do not plan to add or change anything that precludes running Quay via docker.

For the database, however, it's a different story. We really would like to use some features that are hard to port across MySQL and Postgres. Given the traction of the Postgres project, we will likely add features in Quay that do not work (well) on MySQL, at which point these features become unavailable if a MySQL database is used. We may also stop testing MySQL as part of the Quay unit and integration test suite. There is no specific release yet because we have yet to reach the point where a feature only works on Postgres, but if you plan to use Quay for a long-term install, you should do so on Postgres.

**Reply by bhenc** [reply]
Thank you very much for the detailed answer!
